Automattic/wp-calypso
# Seller experience: Make options requirement aware of current site plan

#### Changes proposed in this Pull Request

* Changed the simple path requirement (it's not free)
* Show a different message when the current site already fulfils the requirement.

#### Testing instructions



* Access `start/setup-site/intent?siteSlug=<SITE SLUG>&flags=seller-experience` (which enables the `seller-experience` feature flag)
* Click the "Start Selling" button
* Click "Continue"
* Check the new view
* With Free plan:
![image](https://user-images.githubusercontent.com/3801502/152010403-a8da6a85-7bd2-42cd-9c6d-456f8245ce1b.png)

* Paid Plan:
![image](https://user-images.githubusercontent.com/3801502/152218012-51a7caad-00b0-4f35-9df8-2e9bf5faa10b.png)

* Business or eCommerce plans:
![image](https://user-images.githubusercontent.com/3801502/152218113-43eed127-9d80-4224-9bc5-47abc08f2459.png)


* The link should take you to the plans page.



Fixes #60710
